### PR TITLE
Add action to create .dmg for osx for new releases

### PR DESCRIPTION
**Pull request by girardof**
Create a Github action to generate a .dmg file containing a .app
created with cargo bundle.

### CODE REVIEW

**Comment by ArturKovacs** [comment]
I'm so excited we're gonna have proper mac support 😀

But let's slow down, I do have some questions:
1, When you open the dmg file, do you see something like [this](https://www.howtogeek.com/wp-content/uploads/2018/08/dmg_3.png)? And if so is the name "Emulsion" and is the logo the Emulsion logo?

2, Let's say that someone has v2.1 installed on their Mac. How do they upgrade when v2.2 comes out? Do they just run the new dmg? And if so is this common knowledge or do you think this should be documented?

File: .github/workflows/release-packages.yml [hunk 1]
```diff
@@ -66,3 +66,28 @@ jobs:
...
+    - name: create .dmg file
+      run: hdiutil create 'Emulsion-OSX.dmg' -ov -volname "EmulsionInstall" -fs HFS+ -srcfolder "target/release/bundle/osx/"
+    - name: Upload Release Asset
+      id: upload-release-asset
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ needs.create_release.outputs.upload_url }} 
+        asset_path: ${{ format('Emulsion-OSX.dmg', env.INSTALLER_NAME) }}
```
Comment (ArturKovacs) [inline .github/workflows/release-packages.yml:91]: I think that this is the same as if you just wrote
```
asset_path: Emulsion-OSX.dmg
```

**Comment by girardof** [comment]
> I'm so excited we're gonna have proper mac support 😀
> 
> But let's slow down, I do have some questions:
> 1, When you open the dmg file, do you see something like [this](https://www.howtogeek.com/wp-content/uploads/2018/08/dmg_3.png)? And if so is the name "Emulsion" and is the logo the Emulsion logo?
> 
> 2, Let's say that someone has v2.1 installed on their Mac. How do they upgrade when v2.2 comes out? Do they just run the new dmg? And if so is this common knowledge or do you think this should be documented?

1. For now the .dmg is basic when opening it this is what I get : 

Screenshot 2020-05-14 at 22 07 38

I need to drag and drop the Emulsion.app to the Application folder to install it. For having a background and link to the Application folder we might have to use something like [this bash script](https://github.com/andreyvit/create-dmg) or a solution like [this one](https://stackoverflow.com/questions/96882/how-do-i-create-a-nice-looking-dmg-for-mac-os-x-using-command-line-tools)

2. For updating usually yes the user have to re run the new .dmg and drag and drop the .app again, this will create a message like this : 

Screenshot 2020-05-14 at 22 12 00

I would say it's a rather common knowledge but this is a tech person point of view. 

Some software shows a popup with a link to the dmg when a new version is created for example virtual box : 
Screenshot 2020-05-14 at 22 17 51

**Comment by ArturKovacs** [comment]
Hmmm I don't want to ask too much from you. That's my greatest concern right now 😀 But I want to get this installer thingy right.

If you tell me what are approximately the image dimensions of for example the Firefox dmg, I'll create a background image with those dimensions, and I'd love if you could then put that background and the `create-dmg` script to for example a `./distribution/macos/` folder and have the workflow create a facy pants dmg.

Would this be okay for you?

**Comment by girardof** [comment]
I understand and it's not too much to ask don't worry. I'm not sure if I will have time to look at it tomorrow but I think I could during the weekend. I will tell you more about the size of the background at this point in time 🙂.

**Comment by ArturKovacs** [comment]
I added a background image based on the resolution of the window from the screen shot you sent earlier. It is on the master, in `distribution/macos` so if you merge the master into this branch it will be available to you.

In case you need a different resolution image to make it look good, feel free to create another one from the svg. If you do that, you are going to need the Fira Sans font installed on your system.

**Comment by girardof** [comment]
Quick update, I tried to create a nice looking .dmg using create-dmg but came across the issue described here : 
https://github.com/andreyvit/create-dmg/issues/72

I'm not sure what to do now :(

**Comment by ArturKovacs** [comment]
Ahh this is baad :(

But this has to be possible, there's no way it's not possible to build a dmg on a CI machine.

I would suggest trying a different script/program. Like [this](https://www.npmjs.com/package/create-dmg) nodejs one, or [this](https://pypi.org/project/dmgbuild/) one written in python.

**Comment by ArturKovacs** [comment]
Just as a proof for this being possible: https://github.com/Zettlr/Zettlr/blob/707a3c6f2fd5898dc6d64f2dd7d120a07bb9ffc7/.github/workflows/build.yml#L100

**Comment by girardof** [comment]
I tried with [the node project](https://www.npmjs.com/package/create-dmg) which gave me this command to run : 
```
create-dmg --background 'distribution/macos/background.png' \
           --window-size '750' '600' \
           --app-drop-link '550' '260' \
           --icon 'Emulsion.app' '210' '260' \
           --eula 'LICENSE.txt' \
           'Emulsion.dmg' 'target/release/bundle/osx/'
```
Locally this works perfectly but for some reason when running it inside a Github action I get this error : 
```
Could not find `600`
```

I'm not familiar with Github actions, is there a reason for it not to accept options with two parameters ?

**Comment by ArturKovacs** [comment]
Hmm. It seems that this program only supports the following options.
```
    --overwrite          Overwrite existing DMG with the same name
    --identity=<value>   Manually set code signing identity (automatic by default)
    --dmg-title=<value>  Manually set DMG title (must be <=27 characters) [default: App name]
```

But that's okay. It's still better than the default window in my opinion. So let's just stick with this nodejs script. So the command is going to be
```
create-dmg 'Emulsion.app'
```

**Comment by girardof** [comment]
Sorry, I got confused between the bash and js projects that where both installed on my machine. I finally managed to create a .dmg. 
I had to do a small hack because `create-dmg` return an error (it can't sign the .dmg), therefore the crete_dmg.sh script return true to avoid the action to stop.

**Comment by ArturKovacs** [comment]
Does the fact that the dmg is not signed interfere in any way with installing or using Emulsion?

**Comment by girardof** [comment]
Screenshot 2020-05-22 at 22 05 30

**Comment by girardof** [comment]
> Does the fact that the dmg is not signed interfere in any way with installing or using Emulsion?

No, the only difference I a warning when installing it but it's common. I think you need an apple developer account to sign your .dmg.

**Comment by ArturKovacs** [comment]
Got it. Well that won't happen soon. Maybe one day...

But this is great! Thanks a whole lot I was completely helpless without you.

**Pull request merged**